activation/validation_test.go

```diff
@@ -541,60 +541,31 @@ func TestVerifyChainDeps(t *testing.T) {
 	signer, err := signing.NewEdSigner()
 	require.NoError(t, err)
 
-	ch := types.NIPostChallenge{
-		Sequence:       1,
-		PrevATXID:      types.EmptyATXID,
-		PublishEpoch:   postGenesisEpoch,
-		PositioningATX: goldenATXID,
-		CommitmentATX:  &goldenATXID,
-	}
-	nipostData := newNIPostWithPoet(t, []byte("00"))
-	invalidAtx := newAtx(ch, nipostData.NIPost, 2, types.Address{})
-	require.NoError(t, SignAndFinalizeAtx(signer, invalidAtx))
-	vInvalidAtx, err := invalidAtx.Verify(0, 1)
-	require.NoError(t, err)
+	invalidAtx := newInitialATXv1(t, goldenATXID)
+	invalidAtx.Sign(signer)
+	vInvalidAtx := toVerifiedAtx(t, invalidAtx)
 	vInvalidAtx.SetValidity(types.Invalid)
 	require.NoError(t, atxs.Add(db, vInvalidAtx))
 
 	t.Run("invalid prev ATX", func(t *testing.T) {
-		ch := types.NIPostChallenge{
-			Sequence:       1,
-			PrevATXID:      vInvalidAtx.ID(),
-			PublishEpoch:   postGenesisEpoch,
-			PositioningATX: goldenATXID,
-			CommitmentATX:  nil,
-		}
-		nipostData = newNIPostWithPoet(t, []byte("01"))
-		atx := newAtx(ch, nipostData.NIPost, 2, types.Address{})
-		require.NoError(t, SignAndFinalizeAtx(signer, atx))
-		vAtx, err := atx.Verify(0, 1)
-		require.NoError(t, err)
-		vAtx.SetValidity(types.Unknown)
+		atx := newChainedActivationTxV1(t, goldenATXID, invalidAtx, goldenATXID)
+		atx.Sign(signer)
+		vAtx := toVerifiedAtx(t, atx)
 		require.NoError(t, atxs.Add(db, vAtx))
 
 		ctrl := gomock.NewController(t)
 		v := NewMockPostVerifier(ctrl)
-		v.EXPECT().Verify(ctx, (*shared.Proof)(atx.NIPost.Post), gomock.Any(), gomock.Any())
+		v.EXPECT().Verify(ctx, gomock.Any(), gomock.Any(), gomock.Any())
```

Comment:
Why the change? It should still verify the `Post` that was part of the atx or not?

```suggestion
		v.EXPECT().Verify(ctx, (*shared.Proof)(atx.NIPost.Post), gomock.Any(), gomock.Any())
```

Reply:
The `atx` doesn't contain `Post`, only the wire type does. Keeping this arg-expectation would require translating from the wire type to `shared.Proof` which I figured is not worth the hassle.

Reply:
but the `atx` in this test is a `wire.ActivationTx1`, so it contains a post and the conversion happens in the actual implementation anyway. A proper conversion is also straight forward (if you want to avoid the implicit cast that might break at some point in the future):

```suggestion
		post := &shared.Proof{
			Pow:     atx.NIPost.Post.Pow,
			Indices: atx.NIPost.Post.Indices,
			Nonce:   atx.NIPost.Post.Nonce,
		}
		v.EXPECT().Verify(ctx, post, gomock.Any(), gomock.Any())
```

Reply:
Yes, as I said above.